atmo/jobs/tasks.py

```diff
@@ -62,54 +62,49 @@ def expire_jobs():
 @celery.task(max_retries=3)
 @celery.autoretry(ClientError)
 def update_jobs_statuses():
-    spark_jobs = SparkJob.objects.all()
+    spark_job_runs = SparkJobRun.objects.all()
 
-    # get the jobs with prior runs
-    spark_jobs_with_active_runs = spark_jobs.active().prefetch_related('runs')
+    # get the active (read: not terminated or failed) job runs
+    active_spark_job_runs = spark_job_runs.active().prefetch_related('spark_job')
     logger.debug(
-        'Updating Spark jobs: %s',
-        list(spark_jobs_with_active_runs.values_list('pk', flat=True))
+        'Updating Spark job runs: %s',
+        list(active_spark_job_runs.values_list('pk', flat=True))
     )
 
     # create a map between the jobflow ids of the latest runs and the jobs
-    jobflow_spark_job_map = {
-        spark_job.latest_run.jobflow_id:
-        spark_job for spark_job in spark_jobs_with_active_runs
-    }
+    spark_job_run_map = {}
+    for spark_job_run in active_spark_job_runs:
+        spark_job_run_map[spark_job_run.jobflow_id] = spark_job_run
+
     # get the created dates of the job runs to limit the ListCluster API call
     provisioner = ClusterProvisioner()
-    runs_created_at = spark_jobs_with_active_runs.datetimes(
-        'runs__created_at', 'day'
-    )
+    runs_created_at = active_spark_job_runs.datetimes('created_at', 'day')
 
     # only fetch a cluster list if there are any runs at all
     updated_spark_job_runs = []
     if runs_created_at:
-        logger.debug('Fetching clusters older than %s', runs_created_at[0])
+        earliest_created_at = runs_created_at[0]
+        logger.debug('Fetching clusters older than %s', earliest_created_at)
```

Comment:
It looks like this might be "Fetching clusters since %s". Older than would be this date and prior, yet the code below queries `created_after` this date.